### ISSUE TITLE
Citation summary

### ISSUE THREAD

**Issue by kaplun** [inspirehep/inspire-next#1304]
Legacy supported a mode where several metrics where computed based on the records matching a query. The same functionality should be available on Labs, possibly implemented as an API entry-point allowing for spitting out JSON.

See: https://inspirehep.net/search?ln=en&p=author%3AB.Storaci.1&of=hcs2

**Comment by kaplun** [comment]
@inspirehep/inspire-dir I guess to these number we should introduce the possibility of excluding citations coming from non published papers (this is a recurrent request at least from one user).

@hoc3426 here it is so we don't forget :)

@tomaszgy I think this is the good use case for Neo4J

**Comment by kaplun** [comment]
@eamonnmag: for reference, following our IRL discussion, here's the old ticket with info:
https://github.com/inveniosoftware/invenio/issues/326

For me: example of what pubmed does: https://europepmc.org/authors/0000-0003-3035-4195

Related to the needed input, @eamonnmag what do you think about:

citing paper ID |date of citation | self citation? | published? | RPP? | cited paper ID | date of cited paper
--------------------|--------------------|-------------------|----------------|----------|-------------------|-----------------------
123 | 2016-04-23 | true | true | false | 456 | 1983-03-23
234 | 2016-05-24 | false | true | true | 456 | 1983-03-23
... | ... | ... | ... | ...| ... | ...

If that would be OK, maybe @tomaszgy can produce a dump of such information for a given author?

**Comment by kaplun** [comment]
I have amended the columns to add whether the cited paper is actually cited, so that we can filter easily the cited papers as required from the current citesummary functionality.

Related to the aggregations such as "`Renowned papers (500+)`" I guess these can be computed browser-side without the need to send this info explicitly in the table, right?

**Comment by eamonnmag** [comment]
Yeah, I need to try this out and see where limitations arise. It may need some modifications, but this looks good to me.

**Comment by kaplun** [comment]
So for the subject, in first approximation we can just pick up one subject at random, in case of multiple ones.

**Comment by eamonnmag** [comment]
Simple starting point.  Need to add more functionality and organise the display properly. Also load in more realistic test data.

![citation_summary](https://cloud.githubusercontent.com/assets/282396/16692252/4726ab0c-4530-11e6-89f3-5a46ac072e79.gif)

**Comment by kaplun** [comment]
BTW We should add 2 more columns: 'data' and 'collaboration paper'

**Comment by hoc3426** [comment]
Here's a summary of things people have asked for over the years: http://www.slac.stanford.edu/~hoc/citesummary.html

**Comment by kaplun** [comment]
@hoc3426 Thanks. Beside the specific granularity for the number of authors (our table was distingushing only between collaboration and non-collaboration papers) I believe the other use cases are all fullfilled by the above table. (I assume PDG could be assimilated to reviews)

**Comment by annetteholtkamp** [comment]
RPP (PDF) should be treated as a singular case and not mixed up with reviews. The issue is about citations to the RPP. RPP authors get an inordinate amount of citations which totally distort their citesummary.

- Annette

**Comment by kaplun** [comment]
@eamonnmag the number of row in this table could amount to the 1M. Would your javascript be able to start rendering something while downloading?

**Comment by eamonnmag** [comment]
@kaplun that would be nice, but no. Unfortunately progressive rendering is not possible here. It would also increase the computation significantly. Since the groups are calculated ahead of time (there is a map-reduce operation across all dimensions), you'd have to recalculate all groups an N time intervals, which would ultimately slow things now. A nice progress indicator can be a better solution for now.
